btree/btree.c

```diff
@@ -2034,12 +2151,628 @@ static void generic_move(struct nd *src, struct nd *tgt,
+	return m0_sm_op_sub(&bop->bo_op, P_CLEANUP, P_DONE);
+}
+static bool check_path(void)
+{
+	/*int total_level = oi->i_used;
+	while(total_level >= 0) {
+		if(!is_valid(l_node->n_addr)) {
+			return false;
+		}
+		if(oi->i_level[total_level].l_seq != oi->i_level[total_level].l_node->n_seq)) {
+			return false;
+		}
+		total_level--;
+	}
+	*/
+	return true;
+}
+
+static bool locked(void)
+{
+	return false;
+}
+
+static int64_t lock_op_init(struct m0_sm_op *bo_op, struct node_op *i_nop, int nxt)
+{
+	//m0_rwlock_write_lock(i_nop->no_tree->t_lock);
+	return nxt;
+}
+
+static void lock_op_unlock(struct node_op *i_nop)
+{
+	//m0_rwlock_write_unlock(i_nop->no_tree->t_lock);
+}
+
+static bool cookie_is_set(struct m0_bcookie *k_cookie)
+{
+	return false;
+}
+
+/*
+* checks if given node is still exists
+*/
+static bool is_node_valid(const struct nd *node)
+{
+	return true;
+}
+#if 0
+/*
+* modify_key() will modify key present at node_slot.s_idx
+* if flag = true, update key to new key (node_slot->s_rec.r_key) else, modify key as empty
+* In case of fix key-size format, we can just overwrite existing key with new key,
+* but, for variable key-size format, we need to delete existing record first,
+* and insert new record with modified key as new key size can be different
+*/
+static void modify_key(struct slot *node_slot, bool flag, struct m0_be_tx *tx)
+{
+	struct m0_btree_rec new_rec;
+	if(flag)
+		new_rec.r_key = node_slot->s_rec.r_key;
+	//memcpy instead new_rec.r_val = node_slot->s_rec.r_val;
+
+	node_del(node_slot->s_node, node_slot->s_idx, tx);
+	node_make(node_slot, tx);
+	//node_rec(node_slot);
+	m0_bufvec_copy(&node_slot->s_rec.r_key.k_data, &new_rec.r_key.k_data,
+		       m0_vec_count(&new_rec.r_key.k_data.ov_vec));
+	m0_bufvec_copy(&node_slot->s_rec.r_val, &new_rec.r_val,
+		       m0_vec_count(&new_rec.r_val.ov_vec));
+	node_done(node_slot, tx, true);
+}
+
+#endif
+
+static void alloc(struct m0_btree_oimpl *oi, int height)
+{
+	oi->i_level = m0_alloc(height * sizeof(struct level));
+}
+
+# if 0
+static int callback_temp(struct m0_btree_rec *rec, int64_t key)
+{
+	struct m0_btree_rec temp_rec;
+	int64_t val = key + 1;
+	void* p_key;
+	void *p_val;
+
+	m0_bcount_t ksize = MAX_KEY_SIZE ;
+	m0_bcount_t vsize = MAX_VAL_SIZE ;
+	p_key = &key;
+	p_val = &val;
+
+	temp_rec.r_key.k_data.ov_vec.v_nr     = 1;
+	temp_rec.r_key.k_data.ov_vec.v_count  = &ksize;
+	temp_rec.r_key.k_data.ov_buf          = &p_key;
+
+	temp_rec.r_val.ov_vec.v_nr           = 1;
+	temp_rec.r_val.ov_vec.v_count        = &vsize;
+	temp_rec.r_val.ov_buf                = &p_val;
+
+	m0_bufvec_copy(&rec->r_key.k_data,
+		       &temp_rec.r_key.k_data,
+		       m0_vec_count(&temp_rec.r_key.k_data.ov_vec));
+	m0_bufvec_copy(&rec->r_val, &temp_rec.r_val,
+		       m0_vec_count(&temp_rec.r_val.ov_vec));
+	return 0;
+}
+#endif
+/*get_tick for insert operation
+static int64_t get_tick_insert(struct m0_btree_op *bop)*/
+static int64_t get_tick_insert(struct m0_sm_op *smop)
```

Comment:
The name should be insert_tick instead of get_tick_insert.

Reply:
I changed it to btree_put_tick as per your suggestion on chat